src/__tests__/waitForElementToBeRemoved.test.js

```diff
@@ -130,30 +131,21 @@ test('waits with custom interval', async () => {
   expect(mockFn).toHaveBeenCalledTimes(4);
 });
 
-test('works with legacy fake timers', async () => {
-  jest.useFakeTimers('legacy');
+test.each([TimerMode.Default, TimerMode.Legacy])(
```

Comment:
we should now test all timers legacy and modern, the same, as they're supposed to work the same

Reply:
Unfortunately this does not work right now because the RN jest setup patches `global.Promise` which [breaks modern timers](https://github.com/facebook/jest/issues/10221).

[This PR](https://github.com/facebook/react-native/pull/29754/files) would need to land first.

An alternative approach is to manually capture and restore the original `global.Promise` as seen [here](https://github.com/sbalay/without_await/commit/64a76486f31bdc41f5c240d28263285683755938). WDYT?

Reply:
I went with the alternative approach